components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CertificateMgtUtils.java

```diff
@@ -185,12 +192,144 @@ public ResponseCode removeCertificateFromTrustStore(String alias) {
         return responseCode;
     }
 
+    /**
+     * Method to update the certificate which matches the given alias.
+     *
+     * @param certificate: The base64 encoded certificate string.
+     * @param alias        : Alias of the certificate that should be retrieved.
+     * @return :
+     */
+    public ResponseCode updateCertificate(String certificate, String alias) throws CertificateManagementException {
+
+        InputStream certificateStream = null;
+        try {
+            File trustStoreFile = new File(TRUST_STORE);
+            localTrustStoreStream = new FileInputStream(trustStoreFile);
+            KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            trustStore.load(localTrustStoreStream, TRUST_STORE_PASSWORD);
+
+            if (trustStore.getCertificate(alias) == null) {
+                log.error("Could not update the certificate. The certificate for alias '" + alias + "' is not found" +
+                        " in the trust store.");
+                return ResponseCode.CERTIFICATE_NOT_FOUND;
+            }
+
+            //Generate the certificate from the input string.
+            byte[] cert = (Base64.decodeBase64(certificate.getBytes(StandardCharsets.UTF_8)));
+            certificateStream = new ByteArrayInputStream(cert);
+
+            if (certificateStream.available() == 0) {
+                log.error("Certificate is empty for the provided alias " + alias);
+                return ResponseCode.INTERNAL_SERVER_ERROR;
+            }
+
+            CertificateFactory certificateFactory = CertificateFactory.getInstance(CERTIFICATE_TYPE);
+            Certificate newCertificate = certificateFactory.generateCertificate(certificateStream);
+            X509Certificate x509Certificate = (X509Certificate) newCertificate;
+
+            if (x509Certificate.getNotAfter().getTime() <= System.currentTimeMillis()) {
+                log.error("Could not update the certificate. The certificate expired.");
+                return ResponseCode.CERTIFICATE_EXPIRED;
+            }
+            // If the certificate is not expired, delete the existing certificate and add the new cert.
+            trustStore.deleteEntry(alias);
+            //Store the certificate in the trust store.
+            trustStore.setCertificateEntry(alias, newCertificate);
+            fileOutputStream = new FileOutputStream(trustStoreFile);
+            trustStore.store(fileOutputStream, TRUST_STORE_PASSWORD);
+        } catch (IOException e) {
+            throw new CertificateManagementException("Error updating certificate.", e);
```

Comment:
will this be logged anywhere?

Reply:
This will be logged in handleInternalServer method in RestApiUtils.
https://github.com/wso2/carbon-apimgt/blob/6.x/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/utils/RestApiUtil.java#L740